tests/integration/stepfunctions/v2/choice_operators/utils.py

```diff
@@ -0,0 +1,98 @@
+import json
+from typing import Any, Final
+
+from localstack.services.stepfunctions.asl.utils.json_path import JSONPathUtils
+from localstack.testing.snapshots.transformer import RegexTransformer
+from localstack.utils.strings import short_uid
+from tests.integration.stepfunctions.templates.choiceoperators.choice_operators_templates import (
+    ChoiceOperatorTemplate as COT,
+)
+from tests.integration.stepfunctions.utils import await_execution_success
+
+TYPE_COMPARISONS: Final[list[tuple[Any, bool]]] = [
+    (None, True),  # 0
+    (None, False),  # 1
+    (0, True),  # 2
+    (0, False),  # 3
+    (0.0, True),  # 4
+    (0.0, False),  # 5
+    (1, True),  # 6
+    (1, False),  # 7
+    (1.1, True),  # 8
+    (1.1, False),  # 9
+    ("", True),  # 10
+    ("", False),  # 11
+    (" ", True),  # 12
+    (" ", False),  # 13
+    ("HelloWorld", True),  # 14
+    ("HelloWorld", False),  # 15
+    ("2012-10-09T19:00:55", True),  # 16
+    ("2012-10-09T19:00:55", False),  # 17
+    ("2012-10-09T19:00:55Z", True),  # 18
+    ("2012-10-09T19:00:55Z", False),  # 19
+    ("2023-02-24", True),  # 20
+    ("2023-02-24", False),  # 21
```

Comment:
might also want to add samples for valid timestamps with a timezone, e.g. `2012-10-09T19:00:55+01:00`